pkg/restmapper/controllerrestmapper.go

```diff
@@ -56,7 +60,22 @@ func (m *ControllerRESTMapper) ResourcesFor(input schema.GroupVersionResource) (
 // RESTMapping identifies a preferred resource mapping for the provided group kind.
 func (m *ControllerRESTMapper) RESTMapping(gk schema.GroupKind, versions ...string) (*meta.RESTMapping, error) {
 	ctx := context.TODO()
-
+	// Since versions is optional string slice, it can be empty. If version is not given, we will iterate all the cached
+	// GV and find the first matching RESTMapping.
+	if len(versions) == 0 {
+		for keyGV := range m.cache.groupVersions {
```

Comment:
I _think_ the logic here should be identical to the logic for RESTMappings except you always return the preferred (first) version.

Reply:
I'll revert the change for ControllerRESTMapper, since it's unrelated to this PR.